metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/native_sql_parser.py

```diff
@@ -41,7 +44,48 @@ def get_tables(native_query: str) -> List[str]:
         logger.debug(f"Type={type(tokens[from_index])}")
         if isinstance(tokens[from_index], sqlparse.sql.Identifier):
             # Split on as keyword and collect the table name from 0th position. strip any spaces
-            tables.append(tokens[from_index].value.split("as")[0].strip())
+            table: str = tokens[from_index].value.split("as")[0].strip()
+
+            if len(table.split(".")) == 1:
+                table = f"{default_schema}.{table}"
+
+            tables.append(table)
+
         from_index = from_index + 1
 
     return tables
+
+
+def _advance_sql_query_parsing(native_query: str, default_schema: str) -> List[str]:
+    """
+    This function uses sqllineage package to parse native sql.
+    It supports normal as well as advance sql query construct.
+    """
+    tables: List[str] = []
+
+    parser = LineageRunner(native_query)
```

Comment:
we have logic around LineageRunner in a ton of places in our codebase

does it make sense to refactor those? I'm also working on some sql parsing stuff, so having a cleanly defined interface without all this copy-pasted code would make my life a lot easier

Reply:
Agree, check this thread as well https://acryldata.slack.com/archives/C02ENS87APK/p1685715808871169?thread_ts=1685715193.359649&cid=C02ENS87APK

So Shall I use that interface in this connector initially and then have a separate PR for all of the connectors?